File: src/main/java/com/emiyaconsulting/dramapopapi/model/Person.java
package com.emiyaconsulting.dramapopapi.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

@Entity
@Table(name = "person")
@ToString
@Getter @Setter @NoArgsConstructor
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "first_name")
    private String firstName;
    
    @NotNull
    @Column(name = "last_name", nullable = false)
    private String lastName;
    
    @Column(name = "pfp_url")
    private String pfpUrl;

    @ToString.Exclude
    @OneToMany(mappedBy = "person", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<DramaPersonRole> castCrew = new HashSet<>();
    
    @Column(name = "date_added") @CreationTimestamp
    private Instant dateAdded;

    @Column(name = "date_modified") @UpdateTimestamp
    private Instant dateModified;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Person person)) return false;

        return id != null && Objects.equals(id, person.id);
    }

    @Override
    public final int hashCode() {
        return getClass().hashCode();
    }
}

File: src/main/kotlin/com/emiyaconsulting/dramapopapi/entity/Person.kt
package com.emiyaconsulting.dramapopapi.entity

import jakarta.persistence.*
import org.hibernate.proxy.HibernateProxy

@Entity
@Table(name = "people")
data class Person (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val personID: Long,

    @Column(name = "first_name", nullable = true)
    val firstName: String?,

    @Column(name= "last_name", nullable = false)
    val lastName: String,

    @Column(name="role", nullable = true)
    val role: String?,

    @ManyToMany(mappedBy = "crew")
    var dramas: MutableSet<Drama>? = mutableSetOf()
) {
    final override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null) return false
        val oEffectiveClass =
            if (other is HibernateProxy) other.hibernateLazyInitializer.persistentClass else other.javaClass
        val thisEffectiveClass =
            if (this is HibernateProxy) this.hibernateLazyInitializer.persistentClass else this.javaClass
        if (thisEffectiveClass != oEffectiveClass) return false
        other as Person

        return personID != null && personID == other.personID
    }

    final override fun hashCode(): Int =
        if (this is HibernateProxy) this.hibernateLazyInitializer.persistentClass.hashCode() else javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(  firstName = $firstName   ,   lastName = $lastName   ,   role = $role )"
    }
}